### PR TITLE
Makes grilles attempt to electrocute people that are launched/thrown at them

### PR DESCRIPTION
**Pull request by Skeletonman0**
[QoL] 
## About the PR 
Currently, electrified grilles dont shock anything thrown at them, meaning you can just bypass them with a chair or a corpse. this PR makes the grilles attempt to shock mobs launched / thrown into them.

I made it activate proc/shock after a mob is launched into it to make this work.

## Why's this needed? 
It makes the grilles try to shock any mobs launched at them, hopefully fixing being able to ignore electrified grilles by just chairflipping into the window or atleast reducing being able to almost safely do that.

## Changelog

```changelog
(u)Skeletonman0
(+)Grilles now try to electrocute mobs thrown or launched at them.
```

### CODE REVIEW

**Comment by Wisemonster** [comment]
Doesn't throwing something at a window with a grill underneath hit the grill?

File: code/obj/grille.dm [hunk 1]
```diff
@@ -277,6 +277,7 @@
 		src.visible_message("<span class='alert'><B>[src] was hit by [AM].</B></span>")
 		playsound(src.loc, 'sound/impact_sounds/Metal_Hit_Light_1.ogg', 100, 1)
 		if (ismob(AM))
+			shock(AM, 100 - (60 - src.material.getProperty("electrical")))  // sure loved people being able to throw corpses into these without any consequences.
```
Comment (pali6) [inline code/obj/grille.dm:280]: Shouldn't this have some check for the grille actually being electrified?

Comment (Skeletonman0) [reply code/obj/grille.dm:280]: i believe theres a check for that in the shock proc itself

**Comment by frawhst** [comment]
Echoing Wisemonster's concern, this will shock people who get launched/flip into windows with grilles under them. Is that intended?

**Comment by Skeletonman0** [comment]
I didnt even think of that happening at the time i made this. its probably a bad idea

**Comment by Skeletonman0** [comment]
decided to do some testing and it doesnt shock you if theres a window there, which is strange because i was sure grilles got hit by thrown objects/mobs.

**Comment by Tarmunora** [comment]
> decided to do some testing and it doesnt shock you if theres a window there, which is strange because i was sure grilles got hit by thrown objects/mobs.

probably changed as part of the cross/bump/etc stuff

**Pull request merged**